### PR TITLE
Handle flashlight page errors

### PR DESCRIPTION
**Pull request by benjaminpkane**
#3333 introduces the shared pagination hook for samples (grid and carousel) `useFlashlightPager`. Error handling for pagination requests is missing in this new hook.

Group datasets do not have defined slices until samples are added, so I've added an e2e test that asserts an expected error from the grid when paginating over the undefined default slice.
```py
import fiftyone as fo
dataset = fo.Dataset("empty-group-dataset")
dataset.add_group_field("group", default="empty")
```
Screen Shot 2023-08-02 at 10 39 10 AM

### CODE REVIEW

File: e2e-pw/src/oss/fixtures/loader.ts [hunk 1]
```diff
@@ -145,8 +150,11 @@ export class OssLoader extends AbstractFiftyoneLoader {
       }
     }
 
-    await page.waitForSelector("[data-cy=flashlight-section]", {
-      state: "visible",
-    });
+    await page.waitForSelector(
```
Comment (manivoxel51) [inline e2e-pw/src/oss/fixtures/loader.ts:153]: nit: I think you can use page.getByTestId('TEST_ID').waitFor({ state: 'visible'}) - not sure if it makes a difference

File: e2e-pw/src/oss/poms/panel.ts [hunk 2]
```diff
@@ -39,12 +39,20 @@ export class PanelPom {
...
-  constructor(private readonly gridPom: PanelPom) {}
+  constructor(private readonly panelPom: PanelPom) {}
 
   async histogramLoaded() {
-    await expect(this.gridPom.distributionContainer()).toBeVisible();
+    await expect(this.panelPom.distributionContainer()).toBeVisible();
+  }
+
+  async hasError() {
+    await expect(this.panelPom.errorBoundary()).toBeVisible();
```
Comment (manivoxel51) [inline e2e-pw/src/oss/poms/panel.ts:56]: nice

**Review by manivoxel51 — APPROVED** [review]
🍨

File: e2e-pw/src/oss/fixtures/loader.ts (hunk 1 above)
Comment (benjaminpkane) [reply e2e-pw/src/oss/fixtures/loader.ts:153]: Right, I saw that too. I hesitated because I thought @sashankaryal had a reason for this approach

Comment (sashankaryal) [reply e2e-pw/src/oss/fixtures/loader.ts:153]: Appreciate the benefit of the doubt, Ben!
Let's just say I was learning Playwright. Feel free to change both of these to `getByTestId::waitFor`

**Pull request merged**